File: python/tests/sum_of_multiplies_test.py
import pytest
from correlation import  calculate_sum_of_multiplies

def test_sum_of_multiplies_simple():
    values_x = [1,2,3]
    values_y = [1,1,1]

    assert calculate_sum_of_multiplies(values_x, values_y) == 6

def test_sum_of_multiplies_simple2():
    values_x = [1,2,3]
    values_y = [2,2,2]

    assert calculate_sum_of_multiplies(values_x, values_y) == 12

def test_sum_of_multiplies_simple3():
    values_x = [1,2,3]
    values_y = [4,5,6]

    assert calculate_sum_of_multiplies(values_x, values_y) == 32

File: cpp/tests/sum_of_multiplies_test.cpp
#include <gtest/gtest.h>
#include "correlation.hpp"

TEST(SumOfMultiplies, TestCase1)
{
    std::vector<double> values_x { 1.0, 2.0, 3.0 };
    std::vector<double> values_y { 1.0, 1.0, 1.0 };

    ASSERT_DOUBLE_EQ(calculate_sum_of_multiplies(values_x, values_y), 6.0);
}

TEST(SumOfMultiplies, TestCase2)
{
    std::vector<double> values_x { 1.0, 2.0, 3.0 };
    std::vector<double> values_y { 2.0, 2.0, 2.0 };

    ASSERT_DOUBLE_EQ(calculate_sum_of_multiplies(values_x, values_y), 12.0);
}

TEST(SumOfMultiplies, TestCase3)
{
    std::vector<double> values_x { 1.0, 2.0, 3.0 };
    std::vector<double> values_y { 4.0, 5.0, 6.0 };

    ASSERT_DOUBLE_EQ(calculate_sum_of_multiplies(values_x, values_y), 32.0);
}
